lisa/features/gpu.py

```diff
@@ -0,0 +1,125 @@
+                f"Distro {self._node.os._get_detect_string(self._node)}"
+                "not supported to install CUDA driver."
+            )
+
+        wget_tool = self._node.tools[Wget]
+        # download the cuda driver at /tmp/
+        wget_tool.get(cuda_repo, "/tmp/", cuda_repo_pkg)
+        # install the cuda driver rpm
+        install_result = linux_os.install_packages(
+            f"/tmp/{cuda_repo_pkg}", signed=False
+        )
+        if install_result.exit_code != 0:
+            raise LisaException(
+                f"Failed to install {cuda_repo_pkg}. stdout: {install_result.stdout}"
+            )
+        else:
+            self._log.info("Sucessfully installed cuda-drivers")
+
+    def install_gpu_dep(self) -> None:
+        uname_tool = self._node.tools[Uname]
+        uname_ver = uname_tool.get_linux_information().uname_version
+
+        # install dependency libraries for redhat and CentOS
+        if isinstance(self._node.os, Redhat):
+            # install the kernel-devel and kernel-header packages
+            package_name = f"kernel-devel-{uname_ver} kernel-headers-{uname_ver}"
+            install_result = self._node.os.install_packages(package_name)
+            if install_result.exit_code != 0:
+                raise LisaException(
+                    f"Failed to install {package_name}."
+                    f" stdout: {install_result.stdout}"
+                )
+            # mesa-libEGL install/update is require to avoid a conflict between
+            # libraries - bugzilla.redhat 1584740
+            package_name = "mesa-libGL mesa-libEGL libglvnd-devel"
+            install_result = self._node.os.install_packages(package_name)
+            if install_result.exit_code != 0:
+                raise LisaException(
+                    f"Failed to install {package_name}."
+                    f" stdout: {install_result.stdout}"
+                )
+            # install dkms
+            package_name = "dkms"
+            install_result = self._node.os.install_packages(package_name, signed=False)
+            if install_result.exit_code != 0:
+                raise LisaException(
+                    f"Failed to install {package_name}. stdout: {install_result.stdout}"
+                )
+
+        # install dependency libraraies for Ubuntu
+        elif isinstance(self._node.os, Ubuntu):
+            package_name = (
+                f"build-essential libelf-dev linux-tools-{uname_ver}"
+                f" linux-cloud-tools-{uname_ver} python libglvnd-dev ubuntu-desktop"
+            )
+            install_result = self._node.os.install_packages(package_name)
+            if install_result.exit_code != 0:
+                raise LisaException(
+                    f"Failed to install {package_name}."
+                    f" stdout: {install_result.stdout}"
+                )
+
+    def install_compute_sdk(self, driver: compute_sdk, version: str) -> None:
+        if driver == compute_sdk.GRID:
```

Comment:
It's better to have a way to detect the driver and version in this method.

Reply:
> It's better to have a way to detect the driver and version in this method.

how to detect? you mean from supportability perspective?
I believe the feature class should not be enforcing those. It should only be providing ways to install the requirements and drivers and stuff. In the test case, we should be handling the logic on supportability?

Reply:
Let's see if there is any hint about uninstalled hardware. I'm afraid of that it's not easy to know it in test cases or platform. How the two value will be provided?

Reply:
> Let's see if there is any hint about uninstalled hardware. I'm afraid of that it's not easy to know it in test cases or platform. How the two value will be provided?

no, I didn't mean knowing about "uninstalled hardware". I mean from the test case, depending on the SKU, distro (basically supportability matrix), we would want to install the specific driver. If the test param combination supports for multiple driver, we would want to test with both installations one by one.

Reply:
Can it have a default value? Some cases may not need to test different versions, so it can test with default version. When we write the test cases, it may be more clear on how it's used.

Reply:
> Can it have a default value? Some cases may not need to test different versions, so it can test with default version. When we write the test cases, it may be more clear on how it's used.

Done